src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -191,6 +192,20 @@ internal FileData CreateFileData(FileDataVersionOne v1FileData)
             return fileData;
         }
 
+        private IDictionary<string, string> BuildHashesDictionary(IList<HashVersionOne> hashes)
+        {
+            if (hashes == null) { return null; }
+
+            var v2Hashes = new Dictionary<string, string>();
+
+            foreach (HashVersionOne v1Hash in hashes)
+            {
+                v2Hashes[Utilities.AlgorithmKindNameMap[v1Hash.Algorithm]] = v1Hash.Value;
```

Comment:
>Algorithm [](start = 63, length = 9)

Is there a mapping for every value of `v1Hash.Algorithm` or do we need `TryGet` logic?

Reply:
I'm sure this isn't correct, filed a bug. Not very high priority item, I think, v1 hash production not a common scenario. [https://github.com/Microsoft/sarif-sdk/issues/1116](https://github.com/Microsoft/sarif-sdk/issues/1116)

---
In reply to: [226736401](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226736401) [](ancestors = 226736401)